Tasks/VsTest/vstest.ts

```diff
@@ -578,11 +613,10 @@ function createRunSettingsForTestImpact(vsVersion: number, settingsFile: string,
     return defer.promise;
 }
 
-function setupRunSettingsFileForTestImpact(vsVersion: number, settingsFile: string): Q.Promise<string> {
+function setupSettingsFileForTestImpact(vsVersion: number, settingsFile: string): Q.Promise<string> {
     var defer = Q.defer<string>();
-    var tiaEnabled = tl.getVariable('tia.enabled');
     var exitErrorMessage = "Error occured while setting in test impact data collector. Continuing...";
-    if (tiaEnabled && tiaEnabled.toLowerCase() == "true") {
+    if (isTiaAllowed()) {
```

Comment:
the response file path shuld be created randomly. also we should not check if the tia response file exist. as it wont be there. it will be created/not created by tiaselector.